### ISSUE TITLE
Realistic towns

### ISSUE THREAD

**Issue by YetiWiz** [OpenRCT2/OpenLoco#140]
This might be a bit more involved (like a longer term goal), but as a city-builder fan it pains me to see these horrible town developments in the game.
Lets look at an example: (just look at this mess)
![town](https://user-images.githubusercontent.com/36667284/36558896-cbeb468a-180c-11e8-8fbf-7c40ba87c4f6.jpg)
First of all, for a "city" with a population of 8k, there are far to many high-rises. I even modified my .dat-s for all buildings to never go obsolete and enabled all building styles to add some variety. High-rises should be reserved for bigger cities and even then, they shouldn't be so common. I think there are some variables in the .dat files already that govern when the building will be build. The problem here is that the game considers a town with a population of 8000 as a metropolis.
These definitions:
Hamlet from 0
Village from 150
Town from 500
City from 2500
Metropolis from 8000
should be adjusted accordingly. Cities also seem to have some kind of invisible territory. One should be able to display it to avoid accidentally building stops in the wrong city. That way you could also spot the "downtown" area as well as the suburbs of the city, which would be more house oriented, rather that office buildings. I prefer more smaller and less bigger cities, rather than all medium big. Currently it's hard to go past 60k even if there are no other towns around.
Another problem is that cities keep building roads in a grid like fashion. One should be able to disallow cities to build roads themselves. Here an example how this ruins a city layout:
![town2](https://user-images.githubusercontent.com/36667284/36558899-d1a156aa-180c-11e8-9526-53a6b3b12cf1.jpg)
Having to fix that stuff is a pain in the ass.
Cities tend to build on any available spot. Say you want to make a street like that:
![town3](https://user-images.githubusercontent.com/36667284/36558910-da1240d8-180c-11e8-80db-b8f0860fb616.jpg)
Well you can enjoy it for a couple of seconds before the city chops down all the trees and digs this whole place up.
To sum it up, the idea is:
 - Change the city size definitions to more realistic values
 - Allow cities to grow even bigger
 - Introduce the concept of houses at the borders and high-rises in the center of the city
 - An option to disallow them to build roads (for individual cities a new save format is probably needed)
 - Option to limit city grow (probably new save format)
 - A way to display city territory
 - More airports per city
 - Disallow cities to build on certain tiles (probably new save format, using .dat-s for this is impractical since you cannot build trees or use non flat terrain)

**Comment by BabyDeltic** [comment]
This is tremendous. A million miles away from implementation, but fantastic all the same. My idea was to give towns a purpose (it's down as another issue), such as a market town, mining town, artizan town, commuter town, etc. I also think that a town should be able to build indestructible buildings (churches, town halls, market squares, etc), and semi-indestructable buildings (stadiums, hotels, cinemas etc.)  If our ideas were combined and implemented, then with new and more varied buildings (here's to hoping) OpenLoco would be able to knock OTTD into a cocked hat!
